src/elements/content-sidebar/ActivitySidebar.js

```diff
@@ -671,26 +698,68 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
      */
     fetchFeedItems(shouldRefreshCache: boolean = false, shouldDestroy: boolean = false) {
         const {
+            activeFeedEntryId,
+            activeFeedEntryType,
             api,
             file,
             features,
             hasReplies: shouldShowReplies,
             hasTasks: shouldShowTasks,
             hasVersions: shouldShowVersions,
         } = this.props;
+        const shouldFetchReplies =
+            shouldRefreshCache &&
+            shouldShowReplies &&
+            activeFeedEntryId &&
+            activeFeedEntryType === FEED_ITEM_TYPE_COMMENT;
         const shouldShowAppActivity = isFeatureEnabled(features, 'activityFeed.appActivity.enabled');
         const shouldShowAnnotations = isFeatureEnabled(features, 'activityFeed.annotations.enabled');
 
         api.getFeedAPI(shouldDestroy).feedItems(
             file,
             shouldRefreshCache,
-            this.fetchFeedItemsSuccessCallback,
+            shouldFetchReplies ? this.fetchRepliesForFeedItems : this.fetchFeedItemsSuccessCallback,
             this.fetchFeedItemsErrorCallback,
             this.errorCallback,
             { shouldShowAnnotations, shouldShowAppActivity, shouldShowReplies, shouldShowTasks, shouldShowVersions },
         );
     }
 
+    fetchRepliesForFeedItems = (feedItems: FeedItems) => {
+        const { activeFeedEntryId } = this.props;
+
+        if (!activeFeedEntryId) {
+            return;
+        }
+
+        const handleError = (error: ElementsXhrError) => {
+            this.fetchFeedItemsErrorCallback(feedItems, [error]);
+        };
+
+        this.getActiveFeedEntryData(feedItems)
+            .then(({ id, type }) => {
+                if (
+                    !id ||
+                    !type ||
+                    this.isActiveEntryInFeed(feedItems, activeFeedEntryId) ||
+                    !this.isItemTypeComment(type)
+                ) {
+                    this.fetchFeedItemsSuccessCallback(feedItems);
+                    return;
+                }
+
+                const parentType: CommentFeedItemType =
+                    type === FEED_ITEM_TYPE_COMMENT ? FEED_ITEM_TYPE_COMMENT : FEED_ITEM_TYPE_ANNOTATION;
+
+                this.getFeedItemsWithReplies(feedItems, id, parentType)
```

Comment:
Can we flatten out this promise to avoid needing the duplicate `.catch`?

```
this.getActiveFeedEntryData(feedItems)
    .then(({ id, type }) => {
        if (
            !id ||
            !type ||
            this.isActiveEntryInFeed(feedItems, activeFeedEntryId) ||
            !this.isItemTypeComment(type)
        ) {
            return Promise.resolve(feedItems);
        }

        const parentType: CommentFeedItemType =
            type === FEED_ITEM_TYPE_COMMENT ? FEED_ITEM_TYPE_COMMENT : FEED_ITEM_TYPE_ANNOTATION;

        return this.getFeedItemsWithReplies(feedItems, id, parentType)
    })
    .then(updatedItems => this.fetchFeedItemsSuccessCallback(updatedItems))
    .catch(error => this.fetchFeedItemsErrorCallback(feedItems, [error]));